prac_09/taxi_simulator.py

```diff
@@ -0,0 +1,55 @@
+from prac_09.taxi import Taxi
+from prac_09.silver_service_taxi import SilverServiceTaxi
+
+MENU = "q)uit, c)hoose taxi, d)rive"
+
+
+def main():
+    """Simulate riding in several types of taxis."""
+    total_cost = 0
+    taxis = [Taxi("Prius", 100),
+             SilverServiceTaxi("Limo", 100, 2),
+             SilverServiceTaxi("Hummer", 200, 4)]
+    print("Lets drive!")
+    print(MENU)
+    current_taxi = None
+    menu_choice = input(">>> ").lower()
+    while menu_choice != "q":
+        if menu_choice == "c":
+            print("Taxis available: ")
+            display_available_taxis(taxis)
+            taxi_choice = int(input("Choose taxi: "))
+            try:
+                current_taxi = taxis[taxi_choice]
+            except IndexError:
+                print("Invalid taxi choice")
+        elif menu_choice == "d":
+            if current_taxi:
+                current_taxi.start_fare()
+                distance = float(input("Drive how far? "))
+                current_taxi.drive(distance)
+                trip_cost = current_taxi.get_fare()
+                print(f"Your {current_taxi.name} trip cost you ${trip_cost:.2f}")
+                total_cost += trip_cost
+            else:
+                print("You need to choose a taxi before you can drive")
+        else:
+            print("Invalid option")
+        print(f"Bill to date: ${total_cost:.2f}")
+        print(MENU)
+        menu_choice = input(">>> ").lower()
+
+    print(f"Total trip cost: ${total_cost:.2f}")
+    print("Taxis are now:")
+    display_available_taxis(taxis)
+
+
+def display_available_taxis(taxis):
+    """Display taxis available to choose from."""
+    taxi_number = 0
+    for taxi in taxis:
```

Comment:
we haven't seen it much but theres a function call "enumerate" whichs simplifies the process of iterating through an iterable while also keeping track of the index position. So you can simplify your code to something similair to mine:
    for i, taxi in enumerate(taxis):
        print(f"{i} - {taxi}")